src/edu/sjsu/fwjs/Value.java

```diff
@@ -83,23 +83,30 @@ public String toString() {
  * Note that a closure remembers its surrounding scope.
  */
 class ClosureVal implements Value {
-    private List<String> params;
+    private static final String[] stringArrayHint = new String[0];
+
+    private String[] params;
     private Expression body;
     private Environment outerEnv;
     /**
      * The environment is the environment where the function was created.
      * This design is what makes this expression a closure.
      */
-    public ClosureVal(List<String> params, Expression body, Environment env) {
+    public ClosureVal(String[] params, Expression body, Environment env) {
         this.params = params;
         this.body = body;
         this.outerEnv = env;
     }
+    public ClosureVal(List<String> params, Expression body, Environment env) {
+        this.params = params.toArray(stringArrayHint);
+        this.body = body;
+        this.outerEnv = env;
+    }
     public String toString() {
```

Comment:
Good call. If you don't mind, I'll do this with the addition of the type checker since at that time we'll start remembering the type of a function (and therefore its closures). For now I've added a TODO.